src/libraries/Microsoft.PowerFx.Core/Public/Values/RecordValue.cs

```diff
@@ -136,10 +158,16 @@ internal FormulaValue GetField(FormulaType fieldType, string fieldName)
         /// Derived classes must override to provide values for fields. 
         /// </summary>
         /// <param name="fieldType">Expected type of the field.</param>
-        /// <param name="fieldName">Name of the field.</param>
+        /// <param name="fieldName">Name of the field.</param>        
         /// <param name="result"></param>
         /// <returns>true if field is present, else false.</returns>
         protected abstract bool TryGetField(FormulaType fieldType, string fieldName, out FormulaValue result);
+        
+        protected virtual Task<(bool Result, FormulaValue Value)> TryGetFieldAsync(FormulaType fieldType, string fieldName, CancellationToken cancellationToken)
```

Comment:
>Task<(bool Result, FormulaValue Value)> TryGetFieldAsync [](http://example.com/codeflow?start=26&length=56)

I get async doesn't allow out parameters -  but just checking if this tuple is a common pattern for dealing with it?

Reply:
We could remove the boolean and consider it's false if the method returns null or an ErrorValue but HostBugNullMismatch test will not work as expected - look at TryGetField in MyBadRecordValue